fdbclient/DatabaseBackupAgent.actor.cpp

```diff
@@ -2114,14 +2115,19 @@ class DatabaseBackupAgentImpl {
 				state Future<UID> destUidFuture = backupAgent->getDestUid(tr, logUid);
 				wait(success(statusFuture) && success(destUidFuture));
 
-				UID destUid = destUidFuture.get();
-				if (destUid.isValid()) {
-					destUidValue = BinaryWriter::toValue(destUid, Unversioned());
-				}
 				int status = statusFuture.get();
 				if (!backupAgent->isRunnable((BackupAgentBase::enumState)status)) {
 					throw backup_unneeded();
 				}
+				UID destUid = destUidFuture.get();
+				if (destUid.isValid()) {
```

Comment:
Let's put the check/wait for destUID to be filled in behind a new `abortBackup()` argument, such as `bool waitForDestUID`.  I think it is valid to want to abort a DR that will never get a dest UID due to the remote cluster going unavailable after the DR job was inserted into the database but before any of its tasks were able to run.  This is normally a very short time window, but it wouldn't be if the DR agents were offline when the DR job was created.